src/Display.js

```diff
@@ -49,29 +49,36 @@ class Display extends Component {
 
     constructor(props) {
         super()
-        const {itemWidth, itemHeight, total, buffer, paddingLeft, paddingTop, shouldPrerenderAll} = props
+        const {itemWidth, itemHeight, total, buffer, paddingLeft, paddingTop, shouldPrerenderAll, parentHeight, parentWidth} = props
         this.calculator = new GridCalculator({
             itemWidth,
             itemHeight,
             total,
             buffer,
             paddingLeft,
             paddingTop,
-            maxVisibleIndex: shouldPrerenderAll ? total : 0
+            maxVisibleIndex: shouldPrerenderAll ? total : 0,
+            displayHeight: parentHeight || 0,
+            displayWidth: parentWidth || 0
         })
 
         this.state = this.calculator.getState()
     }
 
     componentDidMount() {
+        const {parentHeight, parentWidth} = this.props
         this.updateDisplaySizeInState()
```

Comment:
If you set display's size directly from props, you should not trigger update its size from bounding rect. So I suggest to move it under `if` statement below